topydo/lib/PrettyPrinterFilter.py

```diff
@@ -125,3 +157,120 @@ def filter(self, p_todo_str, _):
                                 p_todo_str)
 
         return p_todo_str
+
+
+class PrettyPrinterBasicPriorityFilter(PrettyPrinterFilter):
+
+    """ Strips the bracked from the priority of the todo item. Add a (blank)
+        space if not priority is specified (to maintain printing alignment)
+    """
+
+    def __init__(self, p_replacement=" "):
+        super(PrettyPrinterBasicPriorityFilter, self).__init__()
+        self.p_replacement = p_replacement
+
+    def filter(self, p_todo_str, _):
+        """ Find the priority """
+        matches = re.search('^(?P<id>\| *\w+\| )?(\((?P<pri>[A-Z])\) )?', p_todo_str)
+        # we are looking for the id string and priority at the beginning of the
+        # line, in the form of:
+        #   | 48| (V) ...
+        #   |h2j| (C) ...
+
+        # there will always be a match (so don't check for it), even if it's
+        # 'nothing', because both match groups are optional
+        if matches.group('id'):
+            """ If we have an id """
+            if matches.group('pri'):
+                """ If we have a priority """
+                return(matches.group('id') + matches.group('pri') + ' ' + p_todo_str[matches.span()[1]:])
+            else:
+                return(matches.group('id') + self.p_replacement + ' ' + p_todo_str[matches.span()[1]:])
+        else:
+            if matches.group('pri'):
+                """ If we have a priority """
+                return(matches.group('pri') + ' ' + p_todo_str[matches.span()[1]:])
+            else:
+                """ No match found """
+                return(self.p_replacement + ' ' + p_todo_str)
+
+
+class PrettyPrinterHumanDatesFilter(PrettyPrinterFilter):
+
+    """ Turns dates to human readable versions. """
+
+    def __init__(self):
+        super(PrettyPrinterHumanDatesFilter, self).__init__()
+
+    def date_from_match(self, matchgroup):
+        """
+        Takes a match group and then returns a date.
+
+        Assumes the matchgroup has matches named 'year', 'month', and 'day'.
+
+        The returned arrow object has hours, minutes, and seconds set to now.
+        """
+        linedate = arrow.now()  # set the time to 'now' so the comparisions work better
+        linedate = linedate.replace(year=int(matchgroup.group('year')))
+        linedate = linedate.replace(month=int(matchgroup.group('month')))
```

Comment:
There's a bug here, by replacing the items one by one.

The bug is that this sequence may cause an invalid (intermediate) date, causing topydo to crash.

* Today is October 30th 2015 (2015-10-30).
* I have an item with creation date 2015-02-09.
* At this point in the code, a date of 2015-02-30 is created, which is an invalid date. Crash.

 Suggestion is to replace them in once call, if that works. Or, if you're sure what you're doing, catch `ValueError`.

Also make sure that this is covered as a test case.